vm/src/format.rs

```diff
@@ -428,6 +428,28 @@ impl FormatSpec {
     }
 
     pub(crate) fn format_int(&self, num: &BigInt) -> Result<String, &'static str> {
+        let precision_error_case: bool = match self.format_type {
+            Some(FormatType::Binary)
+            | Some(FormatType::Character)
+            | Some(FormatType::Decimal)
+            | Some(FormatType::Octal)
+            | Some(FormatType::HexLower)
+            | Some(FormatType::HexUpper)
+            | Some(FormatType::Number) => true,
+            Some(FormatType::String)
+            | Some(FormatType::ExponentUpper)
+            | Some(FormatType::ExponentLower)
+            | Some(FormatType::FixedPointUpper)
+            | Some(FormatType::FixedPointLower)
+            | Some(FormatType::GeneralFormatUpper)
+            | Some(FormatType::GeneralFormatLower)
+            | Some(FormatType::Percentage) => false,
+            None => true,
+        };
+        if precision_error_case && self.precision.is_some() {
+            return Err("Precision not allowed in integer format specifier");
+        }
+
```

Comment:
How about add simple method and replace original `Ok(magnitude.to_str_radix(~~))`. Like this
```rust
#[inline]
fn format_int_internal(&self, magnitude: BigInt, radix: u32) -> Result<String, &'static str>  {
    match self.precision {
        Some(_) => Err("Precision not allowed in integer format specifier"),
        None => Ok(magnitude.to_str_radix(radix))
    }
}
```
And 
```rust
let raw_magnitude_string_result: Result<String, &'static str> = match self.format_type {
            Some(FormatType::Binary) => format_int_internal(magnitude, 2),
            Some(FormatType::Decimal) => format_int_internal(magnitude, 10),
            Some(FormatType::Octal) => format_int_internal(magnitude, 8),
            Some(FormatType::HexLower) => format_int_internal(magnitude, 16),
            Some(FormatType::HexUpper) => {
                let mut result = format_int_internal(magnitude, 16)?;
                result.make_ascii_uppercase();
                Ok(result)
            }
```